### PR TITLE
Added "openstack esi switch list" command

### PR DESCRIPTION
**Pull request by skanthed**

### CODE REVIEW

File: esiclient/tests/unit/v1/test_switch.py [hunk 1]
```diff
@@ -263,6 +263,62 @@ def mock_gfnifp_call(np, n_dict, client):
...
+                                      'switch_id': 'e4:c7:22:c0:0b:69'},
+            "internal_info": {'tenant_vif_port_id': 'neutron_port_uuid_1'},
+        })
+        self.port2 = utils.create_mock_object({
+            "uuid": "port_uuid_2",
+            "node_uuid": "11111111-2222-3333-4444-bbbbbbbbbbbb",
+            "local_link_connection": {'switch_info': 'switch1',
+                                      'switch_id': 'e4:c7:22:c0:0b:69'},
+            "internal_info": {},
+        })
```
Comment (tzumainn) [inline esiclient/tests/unit/v1/test_switch.py:285]: Super good that you are testing duplicate switches!

File: esiclient/v1/switch.py [hunk 2]
```diff
@@ -124,6 +124,32 @@ def take_action(self, parsed_args):
...
+    def get_parser(self, prog_name):
+        parser = super(ListSwitches, self).get_parser(prog_name)
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        ironic_client = self.app.client_manager.baremetal
+
+        ports = list((port for port in ironic_client.port.list(detail=True)))
```
Comment (tzumainn) [inline esiclient/v1/switch.py:141]: Would ``ports = ironic_client.port.list(detail=True)`` also work? It feels like you are iterating through a list in order to create a list.

File: esiclient/v1/switch.py [hunk 3]
```diff
@@ -124,6 +124,32 @@ def take_action(self, parsed_args):
...
+
+        data = []
+        for port in ports:
+            switch = port.local_link_connection.get('switch_info')
+            switch_id = port.local_link_connection.get('switch_id')
+
+            if [switch, switch_id] not in data:
+                data.append([switch, switch_id])
+        return ["Switch Name", "Switch Id"], data
```
Comment (tzumainn) [inline esiclient/v1/switch.py:150]: `Id` should be `ID` (both letters capitalized)

**Review by tzumainn — CHANGES_REQUESTED** [review]
Looks pretty good! A few minor changes requested.

File: esiclient/v1/switch.py (hunk 3 above)
Comment (skanthed) [reply esiclient/v1/switch.py:150]: Done!

File: esiclient/tests/unit/v1/test_switch.py (hunk 1 above)
Comment (skanthed) [reply esiclient/tests/unit/v1/test_switch.py:285]: Thanks!

File: esiclient/v1/switch.py (hunk 2 above)
Comment (skanthed) [reply esiclient/v1/switch.py:141]: yes totally agree and it worked!

**Comment by skanthed** [comment]
@tzumainn Updated with the latest change!

File: setup.cfg [hunk 4]
```diff
@@ -38,6 +38,7 @@ openstack.esiclient.v1 =
     esi_node_volume_attach = esiclient.v1.node_volume:Attach
     esi_switch_vlan_list = esiclient.v1.switch:ListVLAN
     esi_switch_port_list = esiclient.v1.switch:ListSwitchPort
+    esi_switch_list = esiclient.v1.switch:ListSwitches
```
[the remark this answers is not in the document]
Comment (skanthed) [reply setup.cfg]: @tzumainn updated with the change.

**Review by tzumainn — APPROVED** [review]
Looks good, thanks!

**Pull request merged**